### PR TITLE
[Merged by Bors] - Docs: new usage guide and index page

### PR DESCRIPTION
**Pull request by fhennig**
# Description

*Please add a description here. This will become the commit message of the merge request later.*

## Definition of Done Checklist

- Not all of these items are applicable to all PRs, the author should update this template to only leave the boxes in that are relevant
- Please make sure all these things are done and tick the boxes
 
```[tasklist]
# Author
- [ ] Changes are OpenShift compatible
- [ ] CRD changes approved
- [ ] Helm chart can be installed and deployed operator works
- [ ] Integration tests passed (for non trivial changes)
```

```[tasklist]
# Reviewer
- [ ] Code contains useful comments
- [ ] (Integration-)Test cases added
- [ ] Documentation added or updated
- [ ] Changelog updated
- [ ] Cargo.toml only contains references to git tags (not specific commits or branches)
```

```[tasklist]
# Acceptance
- [ ] Feature Tracker has been updated
- [ ] Proper release label has been added
```

Once the review is done, comment `bors r+` (or `bors merge`) to merge. [Further information](https://bors.tech/documentation/getting-started/#reviewing-pull-requests)

### CODE REVIEW

File: docs/modules/spark-k8s/pages/usage-guide/index.adoc [hunk 1]
```diff
@@ -0,0 +1,3 @@
+= Usage guide
+
+Learn how to load your own xref:usage-guide/job-dependencies.adoc[] or configure an xref:usage-guide/s3.adoc[S3 connection]. Have a look at the xref:usage-guide/examples.adoc[] to learn more about different operatoring modes.
```
Comment (razvan) [inline docs/modules/spark-k8s/pages/usage-guide/index.adoc:3]:
```suggestion
Learn how to load your own xref:usage-guide/job-dependencies.adoc[] or configure an xref:usage-guide/s3.adoc[S3 connection] to access data. Have a look at the xref:usage-guide/examples.adoc[] to learn more about different usage scenarios.
```

**Review by razvan — COMMENTED** [review]
Some ideas for the diagram:
- remove the ...-driver-pod-spec-conf-map since it's not created by the operator
- remove the ...-submit-job config map. It's only needed for logging of the `spark-submit` output. Rather irrelevant for a first high level diagram.
- maybe use a different shapes (not just colors) for different objects and get rid ot the large ServiceAccount or ConfigMap titles.
- maybe also integrate the history server with the S3 bucket for the event files.

**Comment by fhennig** [comment]
I've cleaned up the diagram and added more info on the history server. @razvan

**Review by razvan — APPROVED** [review]
nice

**Comment by fhennig** [comment]
bors merge

**Pull request closed without merging**